aws/rust-runtime/aws-config/src/profile/app_name.rs

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+//! Load an app name from an AWS profile
+
+use crate::provider_config::ProviderConfig;
+use aws_sdk_sts::AppName;
+use aws_types::os_shim_internal::{Env, Fs};
+
+/// Loads an app name from a profile file
+///
+/// This provider will attempt to shared AWS shared configuration and then read the
+/// `sdk-ua-app-id` property from the active profile.
+///
+/// # Examples
+///
+/// **Loads "my-app" as the app name**
+/// ```ini
+/// [default]
+/// sdk-ua-app-id = my-app
+/// ```
+///
+/// **Loads "my-app" as the app name _if and only if_ the `AWS_PROFILE` environment variable
+/// is set to `other`.**
+/// ```ini
+/// [profile other]
+/// sdk-ua-app-id = my-app
+/// ```
+///
+/// This provider is part of the [default app name provider chain](crate::default_provider::app_name).
+#[derive(Debug, Default)]
+pub struct ProfileFileAppNameProvider {
+    fs: Fs,
+    env: Env,
+    profile_override: Option<String>,
+}
+
+impl ProfileFileAppNameProvider {
+    /// Create a new [ProfileFileAppNameProvider}
+    ///
+    /// To override the selected profile, set the `AWS_PROFILE` environment variable or use the [`Builder`].
+    pub fn new() -> Self {
+        Self {
+            fs: Fs::real(),
+            env: Env::real(),
+            profile_override: None,
+        }
+    }
+
+    /// [`Builder`] to construct a [`ProfileFileRegionProvider`]
+    pub fn builder() -> Builder {
+        Builder::default()
+    }
+
+    /// Parses the profile config and attempts to find an app name.
+    pub async fn app_name(&self) -> Option<AppName> {
+        let profile = super::parser::load(&self.fs, &self.env)
+            .await
+            .map_err(|err| tracing::warn!(err = %err, "failed to parse profile"))
+            .ok()?;
+        let selected_profile_name = self
+            .profile_override
+            .as_deref()
+            .unwrap_or_else(|| profile.selected_profile());
+        let selected_profile = profile.get_profile(selected_profile_name)?;
+        selected_profile
+            .get("sdk-ua-app-id")
+            .map(|name| match AppName::new(name.to_owned()) {
+                Ok(app_name) => Some(app_name),
+                Err(err) => {
+                    tracing::warn!(err = %err, "`sdk-ua-app-id` property in profile {} was invalid", selected_profile_name);
```

Comment:
```suggestion
                    tracing::warn!(err = %err, "`sdk-ua-app-id` property in profile `{}` was invalid", selected_profile_name);
```

just to make the error message a little clearer